napari/_vispy/layers/image.py

```diff
@@ -107,6 +107,8 @@ def _on_display_change(self, data=None):
 
         self.node.parent = parent
         self.node.order = self.order
+        for overlay in self.overlays.values():
+            overlay.node.parent = self.node
```

Comment:
```suggestion
        for overlay_visual in self.overlays.values():
            overlay_visual.node.parent = self.node
```

For consistency of names